plugins/PagePerformance/javascripts/jqplotStackedBarEvolutionGraph.js

```diff
@@ -0,0 +1,218 @@
+                    tickColor: this.tickColor
+                }
+            };
+
+            defaultParams.seriesDefaults = {
+                renderer: $.jqplot.BarRenderer,
+                rendererOptions: {
+                    shadowOffset: 1,
+                    shadowDepth: 2,
+                    shadowAlpha: .2,
+                    fillToZero: true,
+                    barMargin: this.data[0].length > 10 ? 2 : 10
+                }
+            };
+
+            defaultParams.stackSeries = true;
+
+            var overrideParams = {
+                legend: {
+                    show: false
+                },
+                canvasLegend: {
+                    show: true
+                }
+            };
+            this.jqplotParams = $.extend(true, {}, defaultParams, this.jqplotParams, overrideParams);
+        },
+
+        _bindEvents: function () {
+            JqplotGraphDataTablePrototype._bindEvents.call(this);
+
+            var self = this;
+            var lastTick = false;
+
+            $('#' + this.targetDivId)
+                .on('jqplotMouseLeave', function (e, s, i, d) {
+                    $(this).css('cursor', 'default');
+                    JqplotGraphDataTablePrototype._destroyDataPointTooltip.call(this, $(this));
+                })
+                .on('jqplotClick', function (e, s, i, d) {
+                    if (lastTick !== false && typeof self.jqplotParams.axes.xaxis.onclick != 'undefined'
+                        && typeof self.jqplotParams.axes.xaxis.onclick[lastTick] == 'string') {
+                        var url = self.jqplotParams.axes.xaxis.onclick[lastTick];
+
+                        broadcast.propagateNewPage(url);
+                    }
+                })
+                .on('jqplotPiwikTickOver', function (e, tick) {
+                    lastTick = tick;
+                    var label;
+
+                    var dataByAxis = {};
+                    var totalValue = 0;
+                    for (var d = 0; d < self.data.length; ++d) {
+                        var valueUnformatted = self.data[d][tick];
+                        if (typeof valueUnformatted === 'undefined' || valueUnformatted === null) {
+                            continue;
+                        }
+
+                        totalValue += valueUnformatted;
+
+                        var axis = self.jqplotParams.series[d]._xaxis || 'xaxis';
+                        if (!dataByAxis[axis]) {
+                            dataByAxis[axis] = [];
+                        }
+
+                        var value = self.formatY(valueUnformatted, d);
+                        var series = self.jqplotParams.series[d].label;
+
+                        var seriesColor = self.jqplotParams.seriesColors[d];
+
+                        dataByAxis[axis].push('<span class="tooltip-series-color" style="background-color: ' + seriesColor + ';"/>' + '<strong>' + value + '</strong> ' + piwikHelper.htmlEntities(series));
```

Comment:
I suppose the `value` here and below is a number but may still be code to convert it to html entities or so just to be extra safe? Also `seriesColor` etc.

Reply:
Besides the total value calculation it's the same as for evolution chart. So should be save...

https://github.com/matomo-org/matomo/blob/3f26e785f015d30d0aeea66aaf7484111b0dbfa9/plugins/CoreVisualizations/javascripts/jqplotEvolutionGraph.js#L101-L118